lib/src/search_result/search-result.dart

```diff
@@ -56,18 +55,22 @@ class SearchResult {
   int fetched = 0;
   int total = 0;
 
+  @override
   Future<SearchResult> next() async {
     if (fetched >= total) {
       return null;
     }
 
     if (request.scroll != null && request.scroll.isNotEmpty) {
-      return await kuzzle
-          .query(KuzzleRequest(
+      final query = KuzzleRequest(
         controller: controller,
-        action: scrollAction,
         scrollId: response.result['scrollId'] as String,
-      ))
+      );
+      if (searchAction != 'searchRoles') {
```

Comment:
bad practice: the parent class is not supposed to know anything about inheriting classes

this should be handled by an override in the role's child class